### PR TITLE
Adds export-key implementation

### PR DESCRIPTION
**Pull request by alex-miller-0**
This PR implements [EXPORT KEY](https://dev.status.im/keycard_api/apdu_exportkey.html)

* Adds command to pass `p1`, `p2`, and `path` array to the card applet
* Adds interface to said command, which splits parametrization into more logical booleans

### CODE REVIEW

File: command_set.go [hunk 1]
```diff
@@ -236,6 +236,36 @@ func (cs *CommandSet) DeriveKey(path string) error {
 	return cs.checkOK(resp, err)
 }
 
+func (cs *CommandSet) ExportKey(derive bool, makeCurrent bool, onlyPublic bool, path string) ([]byte, error) {
+	var p1 uint8
+	if (derive == false) {
+		p1 = 0x00
```
Comment (gravityblast) [inline command_set.go]: can we create some constants for these values so that they are used in `NewCommandExportKey` too?

File: command_set.go [hunk 2]
```diff
@@ -236,6 +236,36 @@ func (cs *CommandSet) DeriveKey(path string) error {
...
+	} else if (makeCurrent == false) {
+		p1 = 0x01
+	} else {
+		p1 = 0x02
+	}
+	var p2 uint8
+	if (onlyPublic == true) {
+		p2 = 0x01
+	} else {
+		p2 = 0x00
```
Comment (gravityblast) [inline command_set.go]: also here it would be good if we have some constants

File: command_set.go [hunk 3]
```diff
@@ -236,6 +236,36 @@ func (cs *CommandSet) DeriveKey(path string) error {
...
+	}
+	cmd, err := NewCommandExportKey(p1, p2, path)
+	if err != nil {
+		return nil, err
+	}	
+
+	resp, err := cs.sc.Send(cmd)
+	// fmt.Printf("res.Data: %x\n", resp.Data)
```
Comment (gravityblast) [inline command_set.go]: comment left, I think we can remove it

File: commands.go [hunk 4]
```diff
@@ -207,6 +208,41 @@ func NewCommandDeriveKey(pathStr string) (*apdu.Command, error) {
...
+//		0x00: return public and private key pair
+//		0x01: return only the public key
+//  @param {pathStr}
+//		Derivation path of format "m/x/x/x/x/x", e.g. "m/44'/0'/0'/0/0"
+func NewCommandExportKey(p1 uint8, p2 uint8, pathStr string) (*apdu.Command, error) {
+	
+	// Choose to derive based on the value of p1
+	data := new(bytes.Buffer)
+	if (p1 == 0x01 || p1 == 0x02) {
+		_, path, err := derivationpath.Decode(pathStr)
```
Comment (gravityblast) [inline commands.go]: depending on the direction of the path, we can derive the key from master, parent, or current key.
if we OR the p1 with that value we allow all the possible options. We can use the same logic used in `NewCommandDeriveKey`

**Review by gravityblast — CHANGES_REQUESTED** [review]
thank you @alex-miller-0 , great job! I left a few comments but it looks good!
It would be nice if you can also sign your commits with your GPG key if you can, it's required to be merged.

File: commands.go (hunk 4 above)
Comment (alex-miller-0) [reply commands.go]: Not sure I follow. We are just defining a derivation path here, right? Isn't the rest of the optionality captured in `p1`?

**Comment by alex-miller-0** [comment]
Added constants into `types.go`. Let me know if you want them somewhere else.

Also just added a PGP key. Will sign the next commit

File: commands.go (hunk 4 above)
Comment (gravityblast) [reply commands.go]: sorry I didn't add much context here.
As specified in the docs of EXPORT KEY, if p1 is 0x02 it says:
>P1 = 0x02 derives the path but also changes the current path of the card. The source for derivation can be set by OR’ing P1 with the constants defined in the DERIVE KEY command. This allows deriving from master, parent or current.

So with `0x02` is using an absolute path, but `0x02 | P1DeriveKeyFromCurrent` and `0x02 | P1DeriveKeyFromParent` derive starting from current key or parent key.

**Comment by gravityblast** [comment]
@alex-miller-0 we can add them together with the other P1 constants in 
https://github.com/status-im/keycard-go/blob/3d287100c0cc77cbab8fbcf8ab7c2a7bdd933381/commands.go#L29-L42

sorry for asking more changes! :)

File: commands.go (hunk 4 above)
Comment (alex-miller-0) [reply commands.go]: Ooh I missed that! Fixed

**Review by gravityblast — APPROVED** [review]
thank you for the changes, can I merge it?

**Comment by alex-miller-0** [comment]
@gravityblast Yes - please do!

**Pull request merged**